pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -290,13 +290,12 @@ func (ctrl *csiSnapshotSideCarController) createSnapshotWrapper(content *crdv1.V
 		// storage system has responded with an error
 		klog.Infof("createSnapshotWrapper: CreateSnapshot for content %s returned error: %v", content.Name, err)
 		if isCSIFinalError(err) {
-			err = ctrl.removeAnnVolumeSnapshotBeingCreated(content)
-			if err != nil {
-				return nil, fmt.Errorf("failed to remove VolumeSnapshotBeingCreated annotation from the content %s: %q", content.Name, err)
+			if err := ctrl.removeAnnVolumeSnapshotBeingCreated(content); err != nil {
```

Comment:
nit, can you use a different variable name instead of "err" just to be more specific?

Reply:
I guess it is not strictly requited and IMO the more idiomatic way is the one from the PR.

Just for reference that inner error does not interfere with the outer one - https://play.golang.com/p/atqr1V2jEMi

Reply:
the way its currently implemented certainly will work. I was proposing to use a different name to avoid potential fat finger updates as the following will compile and pass when missing the magic ":" :
```
if err = ....; err != nil {
}
```

Reply:
This is a cherry-pick.  We are not supposed to make any changes to a cherry-pick if the code is not broken.